test/lit/passes/remove-unused-brs.wast

```diff
@@ -340,9 +340,115 @@
     )
   )
 
-  ;; CHECK:      (func $restructure-select-no-multivalue (type $0)
+  ;; CHECK:      (func $restructure-br_if-value-redudant-in-block-tail-1 (type $2) (result i32)
+  ;; CHECK-NEXT:  (block $parent (result i32)
+  ;; CHECK-NEXT:   (call $nothing)
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (call $get-i32)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $restructure-br_if-value-redudant-in-block-tail-1 (result i32)
+    ;; The br_if's value is equal to the value right after it, so we can remove it.
+    (block $parent (result i32)
+      (call $nothing)
+      (drop
+        (br_if $parent
+          (i32.const 1)
+          (call $get-i32)
+        )
+      )
+      (i32.const 1)
+    )
+  )
+
+  ;; CHECK:      (func $restructure-br_if-value-redudant-in-block-tail-2 (type $2) (result i32)
+  ;; CHECK-NEXT:  (block $parent (result i32)
+  ;; CHECK-NEXT:   (call $nothing)
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (br_if $parent
+  ;; CHECK-NEXT:     (i32.const 2)
+  ;; CHECK-NEXT:     (call $get-i32)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $restructure-br_if-value-redudant-in-block-tail-2 (result i32)
+    ;; As above, but now the value is different, so we do not optimize
+    (block $parent (result i32)
+      (call $nothing)
+      (drop
+        (br_if $parent
+          (i32.const 2)
+          (call $get-i32)
+        )
+      )
+      (i32.const 1)
+    )
+  )
+
+  ;; CHECK:      (func $restructure-br_if-value-redudant-in-block-tail-3 (type $0) (param $x i32) (result i32)
+  ;; CHECK-NEXT:  (block $parent (result i32)
+  ;; CHECK-NEXT:   (call $nothing)
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (br_if $parent
+  ;; CHECK-NEXT:     (call $get-i32)
+  ;; CHECK-NEXT:     (call $get-i32)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $restructure-br_if-value-redudant-in-block-tail-3 (param $x i32) (result i32)
+    ;; As above, but now the value has effects, so we do not optimize
+    (block $parent (result i32)
+      (call $nothing)
+      (drop
+        (br_if $parent
+          (call $get-i32)
+          (call $get-i32)
+        )
+      )
+      (i32.const 1)
```

Comment:
Please make this identical to the `br_if`'s value. That way, this is testing that *even* if the values are identical, we check the effects. (Otherwise, if we checked identity first, this test would not cover effects; this PR atm doesn't do that, but a future refactoring might.)
```suggestion
      (call $get-i32)
```